### PR TITLE
Feature/nack

### PR DESCRIPTION
**Pull request by necrophonic**
Adding correct ack/nack for messages

### CODE REVIEW

File: app/consumer.py [hunk 1]
```diff
@@ -13,21 +25,34 @@ def __init__(self):
...
+            logger.info("Processed successfully")
 
-            processed_ok = processor.process()
-            if processed_ok:
-                logger.info("Processed successfully", tx_id=processor.tx_id)
-                self.acknowledge_message(basic_deliver.delivery_tag, tx_id=processor.tx_id)
+        except BadMessageError as e:
+            # If it's a bad message then we have to reject it
+            self.reject_message(basic_deliver.delivery_tag, tx_id=processor.tx_id)
+            logger.error("Bad message - rejected", exception=e, delivery_count=delivery_count)
```
Comment (eldeal) [inline app/consumer.py]: log action rather than all text

File: app/processors/common_software_processor.py [hunk 2]
```diff
@@ -22,29 +34,34 @@ def setup_logger(self):
...
 
-        return response.content
+        response = remote_call(endpoint, json=self.survey)
 
-    def deliver_zip(self, zip_contents):
-        folder = self.get_ftp_folder(self.survey)
-        return self.ftp.unzip_and_deliver(folder, zip_contents)
+        if response.status_code == 400 or response.content is None:
+            # There was an issue with the data. This would imply the data was
+            # corrupt or wrong in some way that means it cannot be trabsformed
```
Comment (eldeal) [inline app/processors/common_software_processor.py]: typo on transformed

File: app/processors/common_software_processor.py [hunk 3]
```diff
@@ -22,29 +34,34 @@ def setup_logger(self):
...
+            self.logger.error("Failed to transform", request_url=endpoint, sestatus_code=response.status_code)
```
Comment (eldeal) [inline app/processors/common_software_processor.py]: typo on status_code

File: tests/test_common_software_processor.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+    def setUp(self):
+        survey = json.loads(common_software_survey)
+        self.processor = CommonSoftwareProcessor(logger, survey, ftpconn)
+        self.processor.ftp.unzip_and_deliver = MagicMock(return_value=True)
+
+    def _get_response(self):
+        response = Response()
+        response.status_code = 200
+        response._content = b'Some content'
```
Comment (eldeal) [inline tests/test_common_software_processor.py]: Should the b be there?

Comment (necrophonic) [inline tests/test_common_software_processor.py]: Yup - it's how python does `byte` strings

**Pull request merged**